sql/core/src/main/scala/org/apache/spark/sql/sources/rules.scala

```diff
@@ -119,6 +120,10 @@ private[sql] case class PreWriteCheck(catalog: Catalog) extends (LogicalPlan =>
         // The relation in l is not an InsertableRelation.
         failAnalysis(s"$l does not allow insertion.")
 
+      case logical.InsertIntoTable(t, _, _, _, _) =>
```

Comment:
I found this hard since t place could be any logical operator, it should fail except for LogicalRelation.